src/services/EcoBridge/Socket/SocketBridge.ts

```diff
@@ -0,0 +1,584 @@
+                owner: approvalData.owner
+              })
+            )
+
+            this.store.dispatch(
+              ecoBridgeUIActions.setStatusButton({
+                label: 'Approve',
+                isLoading: false,
+                isError: false,
+                isApproved: false,
+                isBalanceSufficient: true
+              })
+            )
+            return
+          }
+
+          this.store.dispatch(
+            ecoBridgeUIActions.setStatusButton({
+              label: 'Bridge',
+              isLoading: false,
+              isError: false,
+              isApproved: true,
+              isBalanceSufficient: true
+            })
+          )
+        } else {
+          this.store.dispatch(
+            ecoBridgeUIActions.setStatusButton({
+              label: 'Something went wrong',
+              isLoading: false,
+              isError: true,
+              isApproved: false,
+              isBalanceSufficient: false
+            })
+          )
+        }
+      }
+    } catch (e) {
+      this.store.dispatch(
+        ecoBridgeUIActions.setBridgeModalStatus({ status: BridgeModalStatus.ERROR, error: getErrorMsg(e) })
+      )
+    }
+  }
+
+  public getBridgingMetadata = async () => {
+    const requestId = this.store.getState().ecoBridge[this.bridgeId as SocketList].lastMetadataCt
+
+    const helperRequestId = (requestId ?? 0) + 1
+
+    this.store.dispatch(this.actions.requestStarted({ id: helperRequestId }))
+
+    this.store.dispatch(this.actions.setBridgeDetailsStatus({ status: SyncState.LOADING }))
+
+    const { from, to } = this.store.getState().ecoBridge.ui
+
+    if (!from.chainId || !to.chainId || !this._account || !from.address || Number(from.value) === 0) return
+
+    const socketTokens = this.store.getState().ecoBridge.socket.lists[this.bridgeId]
+    const fromNativeCurrency = Currency.getNative(from.chainId)
+
+    let fromTokenAddress = ''
+    let toTokenAddress = ''
+
+    const ETHtoWETH = to.chainId === ChainId.XDAI && from.address === fromNativeCurrency.symbol
+    const WETHtoETH = from.chainId === ChainId.XDAI && from.address === WETH_GNOSIS_ADDRESS
+    const XDAItoDAI = from.chainId === ChainId.XDAI && from.address === fromNativeCurrency.symbol
+    const DAItoXDAI =
+      to.chainId === ChainId.XDAI && [DAI_ETHEREUM_ADDRESS, DAI_ARBITRUM_ADDRESS].includes(from.address.toLowerCase())
+
+    // Overrides
+    if (XDAItoDAI) {
+      fromTokenAddress = SOCKET_NATIVE_TOKEN_ADDRESS
+      toTokenAddress = getDAIAddress(to.chainId) ?? ''
+    }
+
+    if (DAItoXDAI) {
+      fromTokenAddress = from.address
+      toTokenAddress = SOCKET_NATIVE_TOKEN_ADDRESS
+    }
+
+    if (ETHtoWETH) {
+      fromTokenAddress = SOCKET_NATIVE_TOKEN_ADDRESS
+      toTokenAddress = WETH_GNOSIS_ADDRESS
+    }
+
+    if (WETHtoETH) {
+      fromTokenAddress = WETH_GNOSIS_ADDRESS
+      toTokenAddress = SOCKET_NATIVE_TOKEN_ADDRESS
+    }
```

Comment:
do we want to evaluate the other ifs if we find a match?
I assume only 1 of them will match but still, do we want that door open?

Reply:
We will turn it into a function that bypasses these checks

Reply:
Resolved by #900